### ISSUE TITLE
Should `async_std_resolver::resolver_from_system_conf` be `async`?

### ISSUE THREAD

**Issue by thomaseizinger** [hickory-dns/hickory-dns#2089]
The above mentioned function is currently marked `async` but doesn't use `await`:

https://github.com/hickory-dns/hickory-dns/blob/d07455476cfba08a661c09f303af2cfbb8c1a5cc/crates/async-std-resolver/src/lib.rs#L141-L143

When digging deeper, @oblique found out that the underlying file IO with `/etc/resolv.conf` is using blocking access:

https://github.com/hickory-dns/hickory-dns/blob/d07455476cfba08a661c09f303af2cfbb8c1a5cc/crates/resolver/src/system_conf/unix.rs#L30-L39

At the moment, this is kind of the worst of both worlds:

1. We need to deal with a `Future` in upper layers.
2. The resulting future doesn't actually use non-blocking IO.

I think a fix in either direction makes sense although I am leaning slightly towards removing `async` because `/etc/resolv.conf` is likely very small so the blocking IO shouldn't matter much in practice.

**Comment by djc** [comment]
Makes sense to me to make the method synchronous, thanks for the bug report!
